SD_ROOT/wz_mini/etc/init.d/S30patchicamera

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides:
+# Short-Description: Patches the iCamera binary if in self-hosted mode
+# Description:       The stock iCamera binary will reset the network if it is unable to reach the internet. This binary patch will change this behavior.
+### END INIT INFO
+
+. /opt/wz_mini/wz_mini.conf
+
+case "$1" in
+	start)
+
+		echo "#####$(basename "$0")#####"
+		
+		if [[ "$ENABLE_SELFHOSTED_MODE" == "true" ]] ; then
+			# Apply the patch. The script here is idempotent.
+			if /opt/wz_mini/bin/bash /opt/wz_mini/usr/bin/patch_icamera.sh ; then
+				echo "iCamera patch applied successfully."
+				exit 0
+			else
+				# If the patch failed, ensure the iCamera shim script calls the stock version (Eg. in the event of a system upgrade to an unsupported version)
+				echo "iCamera patch failed."
+			fi
+		fi
+
+		# If ENABLE_SELFHOSTED_MODE != "true" or if the patch failed, remove the patch if it exists.
+		if [ -f /opt/wz_mini/usr/bin/iCamera.patched ] ; then
+			echo "Reverting iCamera patch."
+			rm -v /opt/wz_mini/usr/bin/iCamera.patched
+		fi
+		# Ensure iCamera shim script points to /system/bin/iCamera
+		if grep -q iCamera.patched /opt/wz_mini/usr/bin/iCamera ; then
+			sed -i 's/\/opt\/wz_mini\/usr\/bin\/iCamera.patched/\/system\/bin\/iCamera/' /opt/wz_mini/usr/bin/iCamera
+		fi
+
+		;;
```

Comment:
I've only visually inspected this, but it looks like this block always runs, not just when selfhosted mode is disabled / patching failed?